client.go

```diff
@@ -112,11 +118,28 @@ func (t *ClientMaster) ClientGet(args shared.GetArgs, retVal *bool) error {
 		// ERR_NO_KEY is handled here!
 		return err
 	} else {
-		// TODO: Handle ERR_DEP
-		// reply contains shared.Value == key, vectorTimeStamp, serverId, clientId
-		fmt.Println("ClientGet is successful", reply)
+		// Handle ERR_DEP
+		// reply contains shared.Value == val, vectorTimeStamp, serverId, clientId
+		checkHistoryKey := historyKey{key, clientId}
+		checkHistoryValue, ok := clientHistory[checkHistoryKey]
+		if ok {
+			// Compare reply.Ts[reply.ClientId] to checkHistoryValue.clientLogicalClock
+			if reply.Ts[reply.ClientId] < checkHistoryValue.clientLogicalClock {
```

Comment:
I know this is the logic we discussed that day but I feel like something is missing. This was specifically targeted towards the corner case, but we should also check for the entire vector timestamp, ie, we should do this client check only if the latest ts for the key and the current reply.Ts are concurrent, right? Otherwise, there can be inconsistencies. Again, a challenge is how do we maintain the history of timestamps for the key. We can discuss this tomorrow.

Reply:
Added TODO, lets discuss and ill update